lib/azure/armrest/storage_account_service.rb

```diff
@@ -276,26 +276,31 @@ def get_private_images(storage_accounts)
             key = get_account_key(storage_account)
           rescue Azure::Armrest::ApiException
             next # Most likely due to incomplete or failed provisioning.
+          else
+            storage_account.access_key = key
           end
 
-          storage_account.all_blobs(key, configuration.max_threads).each do |blob|
-            next unless File.extname(blob.name).casecmp('.vhd') == 0
-            next unless blob.properties.lease_state.casecmp('available') == 0
-
-            # In rare cases the endpoint will be unreachable. Warn and move on.
-            begin
-              blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
-            rescue Errno::ECONNREFUSED, Azure::Armrest::TimeoutException => err
-              msg = "Unable to collect blob properties for #{blob.name}/#{blob.container}: #{err}"
-              log('warn', msg)
-              next
-            end
-
-            next unless blob_properties.respond_to?(:x_ms_meta_microsoftazurecompute_osstate)
-            next unless blob_properties.x_ms_meta_microsoftazurecompute_osstate.casecmp('generalized') == 0
-
-            mutex.synchronize do
-              results << blob_to_private_image_object(storage_account, blob, blob_properties)
+          storage_account.containers.each do |container|
+            next if container.name =~ /diagnostic/i
```

Comment:
Should we skip if only a partial match? should it be `/^diagnostic$/i`?

Reply:
Yes. The bootdiagnostic containers, for example, are automatically appended with a mix of VM name and guid, e.g. "bootdiagnostics-somevm-xyz123abc".

That said, it looks like I could alter this from `/diagnostic/` to `/^bootdiagnostics/` as I don't see any other diagnostic container names.